### DISCUSSION TITLE
有没有高人能为docker版增加个实时转码的功能？

### DISCUSSION THREAD

**Discussion by rendaa** [AlistGo/alist#6290] [Ideas]
artplayer支持的格式或codec还是有点少，而一些资源是rmvb、mkv的，虽然可以用外部播放器播放，但分享给朋友的时候他们的设备可能需要再安装或购买。这本身不是alist需要处理的问题，但从使用体验角度来说会有些遗憾。是否有高人能帮助为alist那个预览器加个实时转码的功能、直接在那个previewer容器里流化播放一些常见的编码或封装格式呢？对个人的小站来说，不需要支持太多并发访问，一两个足以。

也许很多朋友不需要这功能、或觉得这对设备要求偏高了，请有类似希望的朋友能一同表示些兴趣。

**Comment by pongfcnkl** [comment]
实时转码的话那302功能就无效了 必须开web代理了
你服务器吃得消吗

**Reply by rendaa** [reply]
在服务端进行转码确实需要使流量输入和输出服务端，这的确会给服务端增加负担，不过相较于转码这个更消耗资源的业务来说，转发流量的消耗可能是小巫见大巫了…

其实本来是想在NAS上放个alist、向几个朋友分享一些NAS上或本地网络里的资源，所以没太多想302的好处。另一方面，我不会有太多并发访问的情况，NAS还是能凑合应对一路1080p以下清晰度的转码的。这样强度不很高的要求只是个人才会有的需要，但我猜有类似需要的个人还是有一定数量的。

所以原本想法是alist的docker镜像里带个象阿里云previewer那样的转码服务，能不能用ffmpeg+ffserver完成、然后alist里把这服务配置为一个外部预览器…

**Comment by jklincn** [comment]
相同需求。本来就是从 jellyfin 转过来的，emby/jellyfin 的刮削太无语了，没有刮到的资源无法直接通过文件目录打开。alist 完美符合我的需求，但最近碰到有些资源码率实在太高了，60分钟25GB，看来还是要转码才行，这又得回去了。我是自己家里的nas部署，家里上传大概有 50M 左右。

**Comment by luckylemonade24** [comment]
相同需求，如果在网页在线转码播放基础上，webdav也能弄个虚拟目录映射相同文件，但是转码播放就好了，
